.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Run staticchecks and unit tests
+on:
+  pull_request:
+jobs:
+  run:
+    name: go test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install Go
```

Comment:
Can we reuse steps from the build workflow? Or consolidate these workflows?

Reply:
lets revisit reusable workflows at a later point as it doesnt work as expected